engine/runtime/src/test/scala/org/enso/compiler/test/context/SuggestionBuilderTest.scala

```diff
@@ -2012,14 +1996,31 @@ class SuggestionBuilderTest extends CompilerTest {
         )
       )
     }
+
+    "provide type variants when applicable" in {
+      val code =
+        """type My_Tp
+          |    type Variant_A
+          |    type Variant_B
+          |
+          |foo : My_Tp -> My_Tp
+          |foo arg = arg.do_sth""".stripMargin
+      val module      = code.preprocessModule
+      val suggestions = build(code, module)
+      val fooSuggestion = suggestions.collectFirst {
+        case s: Suggestion.Method if s.name == "foo" => s
+      }
+      val fooArgTp = fooSuggestion.get.arguments(1).reprType
+      fooArgTp shouldEqual Suggestion.TypeRep.Sum(
```

Comment:
Again: fully qualified type name and fully qualified names of constructors. Great.